### PR TITLE
fix Hpc7g cluster template

### PR DESCRIPTION
**Pull request by DaisukeMiyamoto**
This PR fix following issues on Hpc7g cluster set up template `HPC/scripts-setup/hpc7g-ubuntu2004-useast1.yaml`
- ParallelCluster doesn't support custom bootstrap actions in outside S3. So I removed bootstrap actions and IAM permissions for S3.
- use `hpc7g.16xlarge` for compute node instead of c7gn.16xlarge

In addition, to use GCC11 in `HPC/scripts-setup/install-tools-headnode-ubuntu2004.sh`, I add GCC11 installation commands on the top of the script.
I checked running this script completed and successfully build GCC11+OpenMPI in my cluster created from updated yaml file.

--
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by juntangc** [comment]
Thanks for reporting this issue,

I have confirmed that the same OnNodeConfigured script works fine if it is placed under s3, but cloudformation will fail when the script is from a weblink (even though it should work according to this example - https://github.com/aws-samples/aws-parallelcluster-post-install-scripts#aws-parallelcluster-post-install-scripts-). 

The OnNodeConfigured script is required to use gcc-11 for HPC applications.  Running the gcc-11 installation on the headNode is not sufficient because the compute instances (dynamically allocated) will not be able to access the libraries associated with gcc-11.  For the examples provided with this guide, it might work because we use ACfL instead of gcc on the examples. 

In summary, please keep the s3 access and change script to something like:
      Script: s3://<s3_bucket>/install-gcc-11.sh      

Regards,

Jun

**Comment by DaisukeMiyamoto** [comment]
Thank you for your advice.

> Running the gcc-11 installation on the headNode is not sufficient because the compute instances (dynamically allocated) will not be able to access the libraries associated with gcc-11.

I understand.
I will update the PR for using  s3://<s3_bucket>/install-gcc-11.sh in the yaml template and add instruction for uploading `install-gcc-11.sh` to S3.

**Comment by juntangc** [comment]
Thanks for pointing out GdrSupport is no longer required.

GdrSupport (Optional, Boolean)
(Optional) Starting with AWS ParallelCluster version 3.0.2, this setting has no effect. Elastic
Fabric Adapter (EFA) support for GPUDirect RDMA (remote direct memory access) is always
enabled if it's supported by the instance type for the Slurm compute resource and the operating
system.

**Comment by juntangc** [comment]
Please squash the commits for merge.
